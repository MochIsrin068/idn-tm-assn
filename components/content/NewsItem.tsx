import React from "react";
import MainText from "./MainText";
import Date from "./Date";
import { Typography, Image } from "antd";
import { ShareAltOutlined } from "@ant-design/icons";

const { Text } = Typography;

type TPropsNewsItem = {
  data: {
    title: string;
    date: string;
    time: string;
    category: string;
    thumbnail: string;
  };
};

export default function NewsItem({ data }: TPropsNewsItem) {
  return (
    <div className="mb-6 hover:opacity-90">
      <div className="flex justify-between">
        <div className="w-2/3">
          <MainText
            isFix
            secondaryText={
              <div className="flex items-center text-sm">
                <Date dateString={data.date} />
                {`, ${data.time}`}
              </div>
            }
            primaryText={data.title}
          />
        </div>
        <div
          className="w-[160px] h-[96px] rounded-md flex items-end justify-end bg-no-repeat"
          style={{
            background: `url("${data.thumbnail}")`,
            backgroundSize: "cover",
          }}
          data-testid="news-thumbnail-container"
        >
          <div className="bg-white py-1 px-2 rounded-tl-md">
            <Image
              src={"/images/idn-logo.png"}
              alt="thumbnail"
              width={60}
              preview={false}
              data-testid="news-thumbnail"
            />
          </div>
        </div>
      </div>

      <div className="flex items-center justify-between mt-1">
        <Text className="text-gray-400">{data.category}</Text>
        <ShareAltOutlined className="cursor-pointer text-xl font-bold" />
      </div>
    </div>
  );
}
