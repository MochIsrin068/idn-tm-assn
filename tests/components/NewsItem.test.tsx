import React from "react";
import { render, screen } from "@testing-library/react";
import NewsItem from "../../components/content/NewsItem";

const mockProps = {
  data: {
    title: "Ini Kunci untuk Terjun ke NFT, Jangan Asal Ikut-Ikutan!",
    date: "2022-08-25",
    time: "09.10",
    category: "Bisnis",
    thumbnail:
      "https://cdn.eraspace.com/pub/media/mageplaza/blog/post/c/a/cara-membuat-nft.jpg",
  },
};

describe("tests components NewsItem", () => {
  it("Perform snapshots test NewsItem", () => {
    const inputDefaultTree = render(<NewsItem {...mockProps} />);
    expect(inputDefaultTree).toMatchSnapshot();
  });

  it("test text dates rendered on screen", () => {
    render(<NewsItem {...mockProps} />);
    const text = screen.getByText(/August 25, 2022/i);
    expect(text).toBeInTheDocument();
  });

  it("test text times rendered on screen", () => {
    render(<NewsItem {...mockProps} />);
    const text = screen.getByText(/, 09.10/i);
    expect(text).toBeInTheDocument();
  });

  it("test text title rendered on screen", () => {
    render(<NewsItem {...mockProps} />);
    const text = screen.getByText(
      /Ini Kunci untuk Terjun ke NFT, Jangan Asal Ikut-Ikutan!/i
    );
    expect(text).toBeInTheDocument();
  });

  it("test background container thambnail", () => {
    render(<NewsItem {...mockProps} />);
    const container = screen.getByTestId("news-thumbnail-container");
    expect(container).toHaveStyle(
      `background: url(https://cdn.eraspace.com/pub/media/mageplaza/blog/post/c/a/cara-membuat-nft.jpg)`
    );
  });

  it("test text category rendered on screen", () => {
    render(<NewsItem {...mockProps} />);
    const text = screen.getByText(/Bisnis/i);
    expect(text).toBeInTheDocument();
  });

  // optional checker and need query by text becouse that not rendered
  it("test text dates wrong rendered on screen", () => {
    render(<NewsItem {...mockProps} />);
    const text = screen.queryByText(/August 28, 2022/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test text times rendered on screen", () => {
    render(<NewsItem {...mockProps} />);
    const text = screen.queryByText(/, 19.11/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test text title wrong rendered on screen", () => {
    render(<NewsItem {...mockProps} />);
    const text = screen.queryByText(/bahaya nft/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test text category wrong rendered on screen", () => {
    render(<NewsItem {...mockProps} />);
    const text = screen.queryByText(/Health/i);
    expect(text).not.toBeInTheDocument();
  });

  it("test wrong background container thambnail", () => {
    render(<NewsItem {...mockProps} />);
    const container = screen.getByTestId("news-thumbnail-container");
    expect(container).not.toHaveStyle(
      `background: url(https://idn-media-bg123.jpg)`
    );
  });
});
