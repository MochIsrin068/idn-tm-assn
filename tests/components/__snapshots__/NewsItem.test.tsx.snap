// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests components NewsItem Perform snapshots test NewsItem 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-6 hover:opacity-90"
      >
        <div
          class="flex justify-between"
        >
          <div
            class="w-2/3"
          >
            <div
              class="mr-4 "
            >
              <span
                class="ant-typography text-gray-400"
              >
                <div
                  class="flex items-center text-sm"
                >
                  <time
                    datetime="2022-08-25"
                  >
                    August 25, 2022
                  </time>
                  , 09.10
                </div>
              </span>
              <h5
                class="ant-typography !m-0 line-clamp-3"
              >
                Ini Kunci untuk Terjun ke NFT, Jangan Asal Ikut-Ikutan!
              </h5>
            </div>
          </div>
          <div
            class="w-[160px] h-[96px] rounded-md flex items-end justify-end bg-no-repeat"
            data-testid="news-thumbnail-container"
            style="background: url(https://cdn.eraspace.com/pub/media/mageplaza/blog/post/c/a/cara-membuat-nft.jpg); background-size: cover;"
          >
            <div
              class="bg-white py-1 px-2 rounded-tl-md"
            >
              <div
                class="ant-image"
                data-testid="news-thumbnail"
                style="width: 60px;"
              >
                <img
                  alt="thumbnail"
                  class="ant-image-img"
                  src="/images/idn-logo.png"
                  width="60"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex items-center justify-between mt-1"
        >
          <span
            class="ant-typography text-gray-400"
          >
            Bisnis
          </span>
          <span
            aria-label="share-alt"
            class="anticon anticon-share-alt cursor-pointer text-xl font-bold"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="share-alt"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 000-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-6 hover:opacity-90"
    >
      <div
        class="flex justify-between"
      >
        <div
          class="w-2/3"
        >
          <div
            class="mr-4 "
          >
            <span
              class="ant-typography text-gray-400"
            >
              <div
                class="flex items-center text-sm"
              >
                <time
                  datetime="2022-08-25"
                >
                  August 25, 2022
                </time>
                , 09.10
              </div>
            </span>
            <h5
              class="ant-typography !m-0 line-clamp-3"
            >
              Ini Kunci untuk Terjun ke NFT, Jangan Asal Ikut-Ikutan!
            </h5>
          </div>
        </div>
        <div
          class="w-[160px] h-[96px] rounded-md flex items-end justify-end bg-no-repeat"
          data-testid="news-thumbnail-container"
          style="background: url(https://cdn.eraspace.com/pub/media/mageplaza/blog/post/c/a/cara-membuat-nft.jpg); background-size: cover;"
        >
          <div
            class="bg-white py-1 px-2 rounded-tl-md"
          >
            <div
              class="ant-image"
              data-testid="news-thumbnail"
              style="width: 60px;"
            >
              <img
                alt="thumbnail"
                class="ant-image-img"
                src="/images/idn-logo.png"
                width="60"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex items-center justify-between mt-1"
      >
        <span
          class="ant-typography text-gray-400"
        >
          Bisnis
        </span>
        <span
          aria-label="share-alt"
          class="anticon anticon-share-alt cursor-pointer text-xl font-bold"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="share-alt"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 000-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
